service/collector/collector.go

```diff
@@ -74,20 +78,26 @@ func (c *Collector) Describe(ch chan<- *prometheus.Desc) {
 func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	c.logger.Log("level", "debug", "message", "collecting metrics")
 
-	collectFuncs := []func(chan<- prometheus.Metric){
-		c.collectClusterInfo,
-		c.collectTrustedAdvisorChecks,
-		c.collectVPCs,
+	// Get aws clients
+	clients, err := c.getAWSClients()
+	if err != nil {
+		c.logger.Log("level", "error", "message", "could not get aws clients", "error", err.Error())
 	}
 
 	var wg sync.WaitGroup
 
+	collectFuncs := []func(chan<- prometheus.Metric, []awsutil.Clients){
+		c.collectClusterInfo,
+		c.collectAccountsTrustedAdvisorChecks,
+		c.collectAccountsVPCs,
+	}
+
 	for _, collectFunc := range collectFuncs {
```

Comment:
Would something like

```
for _, collectFunc := range collectFuncs {
    for _, client := range clients {
        ...
        collectFunc(ch, client)
        ...
    }
}
```

be a little nicer?

Reply:
My thinking is it keeps the multi-client complexity outside of the individual collector functions

Reply:
This was actually my first approach but it forces us to differentiate collectors which need to handle multiple clients from those who dont.
e.g. `collectClusterInfo` vs `collectTrustedAdvisorChecks`, calling `collectClusterInfo` for every client here does not make any sense.

So I decided to move the multi clients logic inside collectors, because of this.

Reply:
Is `collectClusterInfo` the only collector that would take only one client? I feel like it could be an exception.

Reply:
`collectClusterInfo` do not even need a client.

Reply:
Ah, because it's just grabbing the awsconfigs. Would splitting this up into collectFuncs that need aws clients and collectFuncs that don't need aws clients make sense to you?